var/spack/repos/builtin/packages/go/package.py

```diff
@@ -12,28 +36,39 @@ class Go(Package):
 
     extendable = True
 
-    version('1.5.4', git='https://go.googlesource.com/go', tag='go1.5.4')
     version('1.6.2', git='https://go.googlesource.com/go', tag='go1.6.2')
+    version('1.5.4', git='https://go.googlesource.com/go', tag='go1.5.4')
+    version('1.4.2', git='https://go.googlesource.com/go', tag='go1.4.2')
 
-    variant('test',
-            default=True,
-            description="Run tests as part of build, a good idea but quite"
-            " time consuming")
+    variant('test', default=True, description='Build and run tests as part of the build.')
 
     provides('golang')
 
-    # to-do, make non-c self-hosting compilers feasible without backflips
+    depends_on('git', type='alldeps')
+    # TODO: Make non-c self-hosting compilers feasible without backflips
     # should be a dep on external go compiler
     depends_on('go-bootstrap', type='build')
-    depends_on('git', type='alldeps')
+
+    # NOTE: Older versions of Go attempt to download external files that have
+    # since been moved while running the test suite.  This patch modifies the
+    # test files so that these tests don't cause false failures.
+    # See: https://github.com/golang/go/issues/15694
+    @when('@:1.4.3')
+    def patch(self):
+        test_suite_file = FileFilter(join_path('src', 'run.bash'))
+        test_suite_file.filter(
+            r'^(.*)(\$GOROOT/src/cmd/api/run.go)(.*)$',
+            r'# \1\2\3',
+        )
+
+    @when('@1.5.0:')
+    def patch(self):
+        pass
```

Comment:
This problem has been bugging me a lot lately as well.  I believe that this is a Python constraint, but it might be possible to create a workaround.  I'll see what I can do.